arrow/locales.py

```diff
@@ -2566,22 +2651,23 @@ class MoroccoArabicLocale(ArabicLocale):
 
 
 class IcelandicLocale(Locale):
-    def _format_timeframe(self, timeframe, delta):
-
-        timeframe = self.timeframes[timeframe]
-        if delta < 0:
-            timeframe = timeframe[0]
-        elif delta > 0:
-            timeframe = timeframe[1]
+    def _format_timeframe(self, timeframe: TimeFrames, delta: Union[float, int]) -> str:
+        form = self.timeframes[timeframe]
+        if isinstance(form, tuple):
+            if delta < 0:
+                form = form[0]
+            elif delta > 0:
+                form = form[1]
+            # FIXME: handle when delta is 0
 
-        return timeframe.format(abs(delta))
+        return form.format(abs(delta))  # type: ignore
```

Comment:
We should handle when delta is 0. Otherwise mypy `form` in return line can be `Tuple`